### ISSUE TITLE
Latest version of numpy has deprecated numpy.int, numpy.float

### ISSUE THREAD

**Issue by RaphaelOxcitas** [biopython/biopython#4237]
### Setup
biopython 1.80
numpy 1.21.1

I am reporting a problem with Biopython version, Python version, and operating
system as follows:

```python
import sys; print(sys.version)
import platform; print(platform.python_implementation()); print(platform.platform())
import Bio; print(Bio.__version__)
```

(*Please copy and run the above in your Python, and copy-and-paste the output*)

>>> import sys; print(sys.version)
3.9.15 | packaged by conda-forge | (main, Nov 22 2022, 15:55:03) 
[GCC 10.4.0]
>>> import platform; print(platform.python_implementation()); print(platform.platform())
CPython
Linux-4.18.0-448.el8.x86_64-x86_64-with-glibc2.28
>>> import Bio; print(Bio.__version__)
1.80

### Expected behaviour

(*Please fill this in*)

### Actual behaviour

(*Please fill this in, and provide any exception message in full*)
>>> from Bio.PDB import PDBParser
>>> from Bio.PDB.SASA import ShrakeRupley
>>> p = PDBParser(QUIET=1)
>>> struct = p.get_structure("out", "out.pdb")
>>> sr = ShrakeRupley()
>>> sr.compute(struct, level="S")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/RaphOxcitas/.conda/envs/chemistry/lib/python3.9/site-packages/Bio/PDB/SASA.py", line 206, in compute
    asa_array = np.zeros((n_atoms, 1), dtype=np.int)
  File "/home/RaphOxcitas/.conda/envs/chemistry/lib/python3.9/site-packages/numpy/__init__.py", line 284, in __getattr__
    raise AttributeError("module {!r} has no attribute "
AttributeError: module 'numpy' has no attribute 'int'

### Steps to reproduce

(*Please fill this in *)

**Comment by peterjc** [comment]
Actually the deprecation was in numpy 1.20,

https://numpy.org/doc/stable/release/1.20.0-notes.html#deprecations

It looks like the removal was done in numpy 1.21, and we've been testing on the older versions primarily.

Would you like to make a pull request changing this? Probably to use ``numpy.int_`` and ``numpy.float_`` instead?

**Comment by peterjc** [comment]
This is a duplicate of #4203, fixed in #4204 - please try Biopython 1.81 (current release).

Do you think we should belatedly also add this to the NEWS file for that release?
